backend/app/services/adapters/pipeline_stage.py

```
@@ -667,6 +681,244 @@ def get_team_for_task(
+        # Check if content exceeds max length
+        if len(content) > max_length:
+            logger.info(
+                f"Pipeline context exceeds max length ({len(content)} > {max_length}), "
+                f"summarizing for task {task_id}"
+            )
+            # Summarize using AI (synchronous call)
+            content = self._summarize_context_sync(content, max_length)
+
+        return content
+
+    def _summarize_context_sync(self, content: str, target_length: int = None) -> str:
```

Ruff (0.14.10)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)